tinyexpr.c

```diff
@@ -113,6 +119,20 @@ void te_free(te_expr *n) {
 
 static double pi() {return 3.14159265358979323846;}
 static double e() {return 2.71828182845904523536;}
+static double fac(double a) {/* simplest version of fac */
+    if (a < 0.0)
+        return NAN;
+    if (a > UINT_MAX)
+        return INFINITY;
+    unsigned int ua = (unsigned int)(a);
+    unsigned long int result = 1, i = 1;
+    for (i = 1; i <= ua; i++) {
```

Comment:
As far as I can tell, `i` doesn't actually need to be an `unsigned long`. Since it's never going to be bigger than `ua`, it could just be an `unsigned int`. The cast in the definition of `ua` also seems superfluous.

I'd write the code as
```c
unsigned int ua = a;
unsigned long int result = 1;
for (unsigned int i = 1; i <= ua; i++) {
```